src/directives/badge/badge.directive.ts

```diff
@@ -0,0 +1,174 @@
+import { Directive, ElementRef, HostBinding, Input, OnChanges, OnDestroy, Renderer2, SimpleChanges } from '@angular/core';
+import { ColorService } from '../../services/color/color.service';
+import { ThemeColor } from '../../services/color/theme-color';
+import { ContrastService } from '../accessibility/contrast-ratio/contrast.service';
+
+export type BadgeVerticalPosition = 'above' | 'below';
+
+export type BadgeHorizontalPosition = 'before' | 'after';
+
+export type BadgeSize = 'small' | 'medium' | 'large';
+
+@Directive({
```

Comment:
I notice that hovering over the badge shows the same cursor as the button, it also propagates clicks. I notice material doesn't: https://material.angular.io/components/badge/overview

@donna-marijne  should we change this to match materials behavior?

Reply:
Yes, maybe `pointer-events: none` or something like that? Since the badge might overlap other clickable elements on the page, it should not intercept or block click events.

Reply:
That sounds reasonable! Would we ever want something like a badge click event? Not required for this ticket, but just to consider as `pointer-events: none` would have to be reverted if this was ever the case, can you foresee any problems if this was the case? Probably not, just a thought.

Reply:
Possibly, but I guess in that case we would still use something like `pointer-events: none` for non-clickable badges. There might also be a better way than using `pointer-events`, since I think that would also prevent a user selecting the badge text in order to copy it.